hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -289,6 +289,8 @@ public void refreshTimeline() throws IOException {
           srcRecordsWithCkpt.getRight().getLeft(), metrics, overallTimerContext);
     }
 
+    metrics.updateDeltaStreamerSyncMetrics(System.currentTimeMillis());
```

Comment:
Just confirming that you don't want to emit a metric if there was an exception and it errored out.

Reply:
I would suggest to not update the metrics if an error occurred. Personally i just need a metric that also updates when a sync happens but no commit because of missing new data.
As i understand this patch is wrong as it also updates the metrics if something fails, am i right?

Reply:
If something fails an exception should be raised from this function and the metrics.updateDeltaStreamerSyncMetrics(...) function wont be called.
1. If the sync runs but fails due to exception - no metric would be published.
2. If the sync runs but there is no data to sync - metric will be published. 

If lastSync is not updating then the process failed with an exception.

Reply:
Yes. This patch is okay as long as you can set your alerts to catch - no sync metrics being emitted for a period of time.

Another way to monitor would be to emit a 1 for success and 0 for failure. And it lets you set alerts on a rolling window aggregate. 

I have done both personally. Cant say one is better over other by a lot

Reply:
Thanks for your feedback! If you can confirm that an error throws an exception it works as i intended.
Currently we monitor the `commitTime` metric. The problem being that we cannot distinguish between ingestion stopped working and there are simply no records to ingest.